tf2_ros/include/tf2_ros/buffer_server.h

```diff
@@ -75,20 +78,32 @@ class BufferServer
    */
   template<typename NodePtr>
   BufferServer(
-    const tf2::BufferCoreInterface & buffer,
+    const tf2_ros::Buffer & buffer,
     NodePtr node,
     const std::string & ns,
     tf2::Duration check_period = tf2::durationFromSec(0.01))
   : buffer_(buffer),
     logger_(node->get_logger())
   {
+    rcl_action_server_options_t action_server_ops = rcl_action_server_get_default_options();
+    action_server_ops.result_timeout.nanoseconds = (rcl_duration_value_t)RCL_S_TO_NS(5);
     server_ = rclcpp_action::create_server<LookupTransformAction>(
       node,
       ns,
       std::bind(&BufferServer::goalCB, this, std::placeholders::_1, std::placeholders::_2),
       std::bind(&BufferServer::cancelCB, this, std::placeholders::_1),
-      std::bind(&BufferServer::acceptedCB, this, std::placeholders::_1));
-
+      std::bind(&BufferServer::acceptedCB, this, std::placeholders::_1),
+      action_server_ops
+      );
+
+    service_server_ =
+    rclcpp::create_service<LookupTransformService>(node->get_node_base_interface(), node->get_node_services_interface(), ns, std::bind(
+        &BufferServer::serviceCB, this, std::placeholders::_1,
+        std::placeholders::_2),
+        rmw_qos_profile_services_default,
+        nullptr
+        );
```

Comment:
speaking of services and callbacks - will this service be able to respond to multiple requests at the same time? Especially when using timeouts I think that will be necessary

Reply:
You make an excellent point. I thought that obviously yes, each call will be handled in a thread automatically by the service server but for my test it looks like no :thinking: .... investigating

Reply:
Solved with MultithreadedExecutor and ReentrantCallbackGroup